### PR TITLE
fix: ignore npm alias version range error

### LINKED ISSUE #153: 🐛 Bug: pnpm Aliases is not supported
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

No error with dependency version like `npm:eslint-plugin-i@^2.29.1`

### Actual

```
error  Invalid version range for dependency eslint-plugin-import: npm:eslint-plugin-i@^2.29.1  package-json/valid-package-def
```

### Additional Info

https://pnpm.io/aliases

### PR DESCRIPTION
**Pull request by hyoban**
## PR Checklist

-   [x] Addresses an existing open issue: fixes #153
-   [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

This PR also update vscode settings, because vscode automatically does this.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-	"editor.codeActionsOnSave": { "source.fixAll.eslint": true },
+	"editor.codeActionsOnSave": { "source.fixAll.eslint": "explicit" },
```
Comment (JoshuaKGoldberg) [inline .vscode/settings.json:2]: Ha, this has been happening on all my repos. A little inconvenient. I was planning on doing a general update sweep eventually. This works too.

File: src/rules/valid-package-def.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { createRule } from "../createRule.js";
 // so we disable some other errors here.
 const unusedErrorPatterns = [
 	/^Url not valid/i,
-	/^Invalid version range for .+?: file:/i,
+	/^Invalid version range for .+?: (?:file|npm):/i,
```
Comment (JoshuaKGoldberg) [inline src/rules/valid-package-def.ts:10]: [Testing] Could you add a unit test please? `valid-package-def.test.ts`

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
The source code change looks great to me, just looking for some testing to validate it 😎. Thanks!

**Comment by hyoban** [comment]
Test updated

**Review by JoshuaKGoldberg — APPROVED** [review]
💯 thanks!

**Comment by JoshuaKGoldberg** [comment]
@all-contributors please add @hyoban for code.

> 🤖 Beep boop! This comment was added automatically by [all-contributors-auto-action](https://github.com/marketplace/actions/all-contributors-auto-action).
> Not all contributions can be detected from Git & GitHub alone. Please comment any missing contribution types this bot missed.
> ...and of course, thank you for contributing! 💙

**Pull request merged**